src/sentry/web/frontend/accounts.py

```diff
@@ -175,11 +175,15 @@ def account_settings(request):
 
         form.save()
 
-        # remove previously valid email address
+        # update notification settings for those set to primary email with new primary email
+        options = UserOption.objects.filter(user=user, value=old_email)
```

Comment:
Hmm, does this actually work? `value` isn't a string in the database. It's a `UnicodePickledObjectField`.

Also, this isn't indexed. Not a huge deal, but is there a better way to accomplish this here? The index is on `user, project, key`. It'd be better to query on the keys instead of the value here.